### PR TITLE
Support regions without dedicated seed

### LINKED ISSUE #360: Support regions with no seed
As Gardener now supports worker nodes for regions without dedicated seed (MinimalDistance Policy), we should also show those regions in the dashboard. We should show the regions with seed as recommended regions, and inform the user that the other ones have no dedicated seed but the nearest available will be used.
We need to clarify how we can determine, which policy is set in the gardener (MinimalDistance / SameRegion) as this should also be reflected in the dashboard UI.

See https://github.com/gardener/gardener/pull/917

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
Gardener supports selecting a seed automatically based on a configurable policy. We need to expose this in the dashboard if the feature is enabled.

Screen Shot 2019-05-07 at 13 49 31
Screen Shot 2019-05-07 at 13 49 43
Screen Shot 2019-05-07 at 13 50 02

**Which issue(s) this PR fixes**:
Fixes #360

**Special notes for your reviewer**:

**Release note**:

```noteworthy user
When creating clusters you can now select regions from two categories
- Recommended Regions: API servers in same region as your workers (optimal if you require a low latency)
- [NEW] Supported Regions: API servers in another region than your workers (expect a somewhat higher latency)
```
```action operator
Please set the helm configuration value `frontendConfig.seedCandidateDeterminationStrategy` according to Gardener's `ShoootSeedManager` Admission Controller's `candidateDeterminationStrategy` configuration
```

### CODE REVIEW

**Comment by petersutter** [comment]
the chart has to be adapted for the `seedCandidateDeterminationStrategy` config

**Comment by grolu** [comment]
@vlerenc Please have a look at the screenshots and give me feedback, is this ok / easy to understand?

**Comment by vlerenc** [comment]
How about?
- Recommended Regions (API server in same region)
- Supported Regions (API server in another region)

**Comment by vlerenc** [comment]
And as hint then how about:
- API server in same region as your workers (optimal latency).
- API server in another region than your workers (longer latency; picked by Gardener based on internal considerations, e.g. geographic proximity).

**Comment by petersutter** [comment]
- Use texts provided by @vlerenc 
- Update release notes according to new texts
- custom header (with subtitle)

**Comment by grolu** [comment]
@vlerenc see updated screenshots

**Comment by vlerenc** [comment]
Sorry, when seeing it, a few minor/cosmetic improvements, if I may:

Let's make it plural:
- Recommended Regions (API servers in same region)
- Supported Regions (API servers in another region)

And as for the texts, that you see only one at a time:
- API servers in same region as your workers (recommended, optimal if you require low latency).
- API servers in another region than your workers (supported, expect somewhat higher latency, picked by Gardener based on internal considerations, e.g. geographic proximity).

Hopefully, we can avoid too many questions with these texts and they are self-explanatory enough for the end users to see the impact? I have not said, why we have supported regions/this logic, but this would blow up the texts even more. Hmm... :-/

**Comment by grolu** [comment]
@vlerenc see updated screenshots

**Comment by vlerenc** [comment]
Thank you, only one last/minor change to the texts, but then please just go ahead:
- API servers in same region as your workers (optimal if you require a low latency).
- API servers in another region than your workers (expect a somewhat higher latency, picked by Gardener based on internal considerations such as geographic proximity).

File: frontend/src/dialogs/CreateClusterDialog.vue [hunk 1]
```diff
@@ -582,7 +585,28 @@ export default {
       return get(this.shootDefinition, 'spec.addons', {})
     },
     regions () {
-      return this.regionsByCloudProfileName(this.cloudProfileName)
+      const { regionsWithSeed, regionsWithoutSeed } = this.regionsByCloudProfileName(this.cloudProfileName)
+      const showAllRegions = !isEmpty(this.cfg.seedCandidateDeterminationStrategy) && this.cfg.seedCandidateDeterminationStrategy !== 'SameRegion'
+      const regions = []
+      if (regionsWithSeed.length > 0) {
```
Comment (petersutter) [inline frontend/src/dialogs/CreateClusterDialog.vue]: isEmpty...

File: frontend/src/dialogs/CreateClusterDialog.vue [hunk 2]
```diff
@@ -582,7 +585,28 @@ export default {
       return get(this.shootDefinition, 'spec.addons', {})
     },
     regions () {
-      return this.regionsByCloudProfileName(this.cloudProfileName)
+      const { regionsWithSeed, regionsWithoutSeed } = this.regionsByCloudProfileName(this.cloudProfileName)
```
Comment (petersutter) [inline frontend/src/dialogs/CreateClusterDialog.vue]: rename regions to regionItems

create computed property

select default from regionsWithSeed

File: frontend/src/store/index.js [hunk 3]
```diff
@@ -138,7 +139,10 @@ const getters = {
   regionsByCloudProfileName (state, getters) {
     return (cloudProfileName) => {
       const cloudProfile = getters.cloudProfileByName(cloudProfileName)
-      return uniq(map(get(cloudProfile, 'data.seeds'), 'data.region'))
+      const regionsWithSeed = uniq(map(get(cloudProfile, 'data.seeds'), 'data.region'))
```
Comment (petersutter) [inline frontend/src/store/index.js]: split in two

**Pull request merged**